treeherder/webapp/api/bug.py

```diff
@@ -12,16 +12,31 @@ class BugJobMapViewSet(viewsets.ViewSet):
     def create(self, request, project):
         """Add a new relation between a job and a bug."""
         job_id = int(request.data["job_id"])
-        # TODO: Support either internal reference to a bug or Bugzilla ID
-        bugzilla_id = int(request.data["bug_id"])
         bug_open = bool(request.data["bug_open"])
 
+        bugzilla_id = request.data.get("bug_id")
+        internal_bug_id = request.data.get("internal_id")
+        try:
+            bugzilla_id = bugzilla_id and int(bugzilla_id)
+            internal_bug_id = internal_bug_id and int(internal_bug_id)
+        except TypeError:
+            return Response("bugzilla_id and internal_id must be integers", HTTP_400_BAD_REQUEST)
+        if internal_bug_id is None and bugzilla_id is None:
+            return Response(
+                "At least one of bug_id or internal_id is required", HTTP_400_BAD_REQUEST
```

Comment:
comment doesn't align with if statement.  change this error to be `bug_id and internal_id`, or change the if s/and/or/

Reply:
further thought, wouldn't it be better to ensure we always have an internal_id, bug_id is optional IIRC

Reply:
Sometimes classifications are manual with a bug number and the internal ID is not available to the Treeherder user. The lookup of the internal ID is below.